test_conformance/half/Test_vStoreHalf.cpp

```diff
@@ -562,151 +632,230 @@ int Test_vStoreHalf_private( cl_device_id device, f2h referenceFunc, d2h doubleR
+            if (NULL == kernels[vectorSize][1])
             {
                 gFailCount++;
-                vlog_error( "\t\tFAILED -- Failed to create double private kernel. (%d)\n", error );
+                vlog_error("\t\tFAILED -- Failed to create double private "
+                           "kernel. (%d)\n",
+                           error);
                 return error;
             }
 
-            if(g_arrVecSizes[vectorSize] == 3) {
-                doublePrograms[vectorSize][2] = MakeProgram( device, double_source_local_store_v3, sizeof(double_source_local_store_v3) / sizeof( double_source_local_store_v3[0]) );
-            } else {
-                doublePrograms[vectorSize][2] = MakeProgram( device, double_source_local_store, sizeof(double_source_local_store) / sizeof( double_source_local_store[0]) );
+            if (g_arrVecSizes[vectorSize] == 3)
+            {
+                doublePrograms[vectorSize][2] =
+                    MakeProgram(device, double_source_local_store_v3,
+                                sizeof(double_source_local_store_v3)
+                                    / sizeof(double_source_local_store_v3[0]));
+            }
+            else
+            {
+                doublePrograms[vectorSize][2] =
+                    MakeProgram(device, double_source_local_store,
+                                sizeof(double_source_local_store)
+                                    / sizeof(double_source_local_store[0]));
             }
-            if( NULL == doublePrograms[ vectorSize ][2] )
+            if (NULL == doublePrograms[vectorSize][2])
             {
                 gFailCount++;
                 return -1;
             }
 
-            doubleKernels[ vectorSize ][2] = clCreateKernel( doublePrograms[ vectorSize ][2], "test", &error );
-            if( NULL == kernels[vectorSize][2] )
+            doubleKernels[vectorSize][2] =
+                clCreateKernel(doublePrograms[vectorSize][2], "test", &error);
+            if (NULL == kernels[vectorSize][2])
             {
                 gFailCount++;
-                vlog_error( "\t\tFAILED -- Failed to create double local kernel. (%d)\n", error );
+                vlog_error("\t\tFAILED -- Failed to create double local "
+                           "kernel. (%d)\n",
+                           error);
                 return error;
             }
         }
     } // end for vector size
 
+    const char *reset[] = {
+        "__kernel void reset( __global float *p, __global ushort *f,\n"
+        "                   uint extra_last_thread)\n"
+        "{\n"
+        "   size_t i = get_global_id(0);\n"
+        "   *(f + i) = 0xdeaddead;"
+        "}\n"
+    };
+
+    if (!gHostReset)
+    {
+        resetProgram =
+            MakeProgram(device, reset, sizeof(reset) / sizeof(reset[0]));
+        resetKernel = clCreateKernel(resetProgram, "reset", &error);
+        if (NULL == resetProgram || NULL == resetKernel)
```

Comment:
I think we should check the `NULL == resetProgram` error case before calling `clCreateKernel()` , rather than afterwards, as is the convention in the calls to `MakeProgram` / `clCreateKernel` before this.

Reply:
Done.